source/contracts/reporting/Market.sol

```diff
@@ -124,38 +134,54 @@ contract Market is DelegationTarget, Typed, Initializable, Ownable, IMarket {
+        // The owner gets the no-show REP bond
+        _reputationToken.transfer(owner, _reputationToken.balanceOf(this));
+        // The owner gets the reporter gas costs
+        require(getOwner().call.value(reporterGasCostsFeeAttoeth)());
         return true;
     }
 
-    function disputeDesignatedReport() public triggersMigration returns (bool) {
+    function disputeDesignatedReport(uint256[] _payoutNumerators, uint256 _attotokens) public triggersMigration returns (bool) {
         require(getReportingState() == ReportingState.DESIGNATED_DISPUTE);
         designatedReporterDisputeBondToken = DisputeBondTokenFactory(controller.lookup("DisputeBondTokenFactory")).createDisputeBondToken(controller, this, msg.sender, Reporting.designatedReporterDisputeBondAmount(), tentativeWinningPayoutDistributionHash);
         reportingWindow.getReputationToken().trustedTransfer(msg.sender, designatedReporterDisputeBondToken, Reporting.designatedReporterDisputeBondAmount());
-        updateTentativeWinningPayoutDistributionHash(tentativeWinningPayoutDistributionHash);
+        if (_attotokens > 0) {
+            require(derivePayoutDistributionHash(_payoutNumerators) != tentativeWinningPayoutDistributionHash);
+            IReportingToken _reportingToken = getReportingToken(_payoutNumerators);
+            _reportingToken.trustedBuy(msg.sender, _attotokens);
+        } else {
+            updateTentativeWinningPayoutDistributionHash(tentativeWinningPayoutDistributionHash);
+        }
         reportingWindow.updateMarketPhase();
         return true;
     }
 
-    function disputeLimitedReporters() public triggersMigration returns (bool) {
-        require(getReportingState() == ReportingState.LIMITED_DISPUTE);
-        limitedReportersDisputeBondToken = DisputeBondTokenFactory(controller.lookup("DisputeBondTokenFactory")).createDisputeBondToken(controller, this, msg.sender, Reporting.limitedReportersDisputeBondAmount(), tentativeWinningPayoutDistributionHash);
-        reportingWindow.getReputationToken().trustedTransfer(msg.sender, limitedReportersDisputeBondToken, Reporting.limitedReportersDisputeBondAmount());
+    function disputeFirstReporters(uint256[] _payoutNumerators, uint256 _attotokens) public triggersMigration returns (bool) {
+        require(getReportingState() == ReportingState.FIRST_DISPUTE);
+        firstReportersDisputeBondToken = DisputeBondTokenFactory(controller.lookup("DisputeBondTokenFactory")).createDisputeBondToken(controller, this, msg.sender, Reporting.firstReportersDisputeBondAmount(), tentativeWinningPayoutDistributionHash);
+        reportingWindow.getReputationToken().trustedTransfer(msg.sender, firstReportersDisputeBondToken, Reporting.firstReportersDisputeBondAmount());
         IReportingWindow _newReportingWindow = getUniverse().getNextReportingWindow();
-        return migrateReportingWindow(_newReportingWindow);
+        _newReportingWindow.migrateMarketInFromSibling();
```

Comment:
What made you decide to extract out this function?  In any migration we _must_ add the market to a new window and remove it from its current window and update the reporting window.  It feels like we would decrease the chance of error by packing those 3 operations (which must be atomic) into a function together so someone can't forget one.

Reply:
actually that's a good point. I believe this was not done mindfully. Will add back.